### DISCUSSION TITLE
Differentiate between Warning and Errors

### DISCUSSION THREAD

**Discussion by joncode21** [phcode-dev/phoenix#1792] [General]
Hi, if it is possible in the next update, your should differentiate between Warnings and Errors

Everything in the IDE is considered as an Error, but they can still run.

From experience, if it Errors,  the IDEs should not be able to run. However, warnings still can, like the the side of the IDE ,rather than just a red circle with and X ,replace it with a yellow triangle and an exclamation mark

**Comment by abose** [comment]
@joncode21 which language are you facing this issue? 
1. for HTML/CSS warnings and errors are hadnled as you would expect it to.
2. For JS files, if you have not configured ESLint, then Phoenix code will default to JSHint, which only supports errors. To use ESLint which is more powerful, follow this doc: https://docs.phcode.dev/docs/Features/Problems%20Panel/ESLint/

If you can help with specific examples, i will be able to address this better.

**Reply by joncode21** [reply]
<img width="242" alt="html errors" src="https://github.com/user-attachments/assets/2fef758f-bbab-49ce-ac98-bd7ad3341b36">

But I mean all these are just suggestions. The languages that I am using is PHP combined with HTML and CSS.

As  seen here there is an error shown, what I would suggest instead is that it should only  be a warning  as the overall file can still run. 

From experience with other IDEs , There is a difference between warning and errors. 

1. If  the whole file display full of errors, then the whole file should not be able to run . 

2. If there are warnings,  the file  should be allowed to run even tho it is displayed on the IDE.

**Reply by abose** [reply]
@joncode21 Looks like What you are looking for is customizations in the rules. Please see this [doc](https://docs.phcode.dev/docs/Features/Problems%20Panel/html-lint/) for using a custom `.htmlvalidate.json` file to customize this behaviour.

We follow W3C spec by default; according to the spec, missing `DOCTYPE` is considered an error. You can test it using the [W3C validator](https://validator.w3.org/nu/#textarea). As an editor, we must follow the W3C spec for correctness, and not make our own rules to prevent the legacy IE vs every other browser situation, but for code editors.

The below code will disable the doctype error message.
```json
{
  "extends": ["html-validate:PRESET"],
  "rules": {
       "missing-doctype": "off"
   }
}
```
other rules can be found here: https://html-validate.org/rules/presets.html#comparison

**Reply by abose** [reply]
One thing we can do from our end is to make it easier to disable or enable individual rules in the project. Created an issue to track here: https://github.com/phcode-dev/phoenix/issues/1797

@joncode21 if you have any specific rules, that you find should be a warning/other, please drop a comment here with an example, will evaluate if that should be the default behaviour and needs changing.

**Reply by joncode21** [reply]
I mean becuz I started my coding from Visual Studio 2022 IDE and there were warnings and errors that were obviously displayed at the lines.
